packages/@aws-cdk/aws-glue/lib/table.ts

```diff
@@ -0,0 +1,239 @@
+   *
+   * @default data/
+   */
+  prefix?: string;
+
+  /**
+   * Columns of the table.
+   */
+  columns: Column[];
+
+  /**
+   * Partition columns of the table.
+   *
+   * @default table is not partitioned
+   */
+  partitionKeys?: Column[]
+
+  /**
+   * Storage type of the table's data.
+   */
+  storageType: StorageType;
+
+  /**
+   * Indicates whether the table's data is compressed or not.
+   *
+   * @default false
+   */
+  compressed?: boolean;
+
+  /**
+   * Indicates whether the table data is stored in subdirectories.
+   *
+   * @default false
+   */
+  storedAsSubDirectories?: boolean;
+}
+
+/**
+ * A Glue table.
+ */
+export class Table extends cdk.Construct implements ITable {
+  /**
+   * Creates a Table construct that represents an external table.
+   *
+   * @param scope The scope creating construct (usually `this`).
+   * @param id The construct's id.
+   * @param props A `TableImportProps` object. Can be obtained from a call to `table.export()` or manually created.
+   */
+  public static import(scope: cdk.Construct, id: string, props: TableImportProps): ITable {
+    return new ImportedTable(scope, id, props);
+  }
+
+  public readonly database: IDatabase;
+  public readonly bucket: s3.Bucket;
+  public readonly prefix: string;
+
+  public readonly tableName: string;
+  public readonly tableArn: string;
+
+  public readonly storageType: StorageType;
+  public readonly columns: Column[];
+  public readonly partitionKeys?: Column[];
+
+  constructor(scope: cdk.Construct, id: string, props: TableProps) {
+    super(scope, id);
+    this.database = props.database;
+    this.bucket = props.bucket || new s3.Bucket(this, 'Bucket');
+    this.storageType = props.storageType;
+    this.prefix = props.prefix || 'data/';
+    this.columns = props.columns;
+    this.partitionKeys = props.partitionKeys;
+
+    const tableResource = new CfnTable(this, 'Table', {
+      catalogId: props.database.catalogId,
+
+      databaseName: props.database.databaseName,
+
+      tableInput: {
+        name: props.tableName,
+        description: props.description || `${props.tableName} generated by CDK`,
+
+        partitionKeys: renderColumns(props.partitionKeys),
+
+        storageDescriptor: {
+          location: cdk.Fn.join('', ['s3://', this.bucket.bucketName, '/', this.prefix]),
+          compressed: props.compressed === undefined ? false : props.compressed,
+          storedAsSubDirectories: props.storedAsSubDirectories === undefined ? false : props.storedAsSubDirectories,
+          columns: renderColumns(props.columns),
+          inputFormat: props.storageType.inputFormat,
+          outputFormat: props.storageType.outputFormat,
+          serdeInfo: {
+            serializationLibrary: props.storageType.serializationLibrary
+          }
+        },
+
+        tableType: 'EXTERNAL_TABLE'
+      }
+    });
+
+    this.tableName = tableResource.ref;
+    this.tableArn = cdk.Fn.join('', [this.database.databaseArn, '/', this.tableName]);
```

Comment:
Use `formatArn`?

Reply:
Can you provide an example? I don't see how it's useful here.

Reply:
Forget about it... But still, no need to use `Fn.join` explicitly, use normal string concatenation (much more readable and less leaky):

```ts
this.tableArn = `${this.database.databaseArn}/${this.tableName}`
```